File: Cpp/1424_Diagonal_Traverse_II.cpp
class Solution {
public:
    vector<int> findDiagonalOrder(vector<vector<int>>& nums) {
        vector<int> result;
        vector<vector<int>> sum;
        for(int i=nums.size()-1; i>=0; --i){
            for(int j=0; j<nums[i].size(); ++j){
                if(i+j>=sum.size()) sum.resize(i+j+1);
                sum[i+j].push_back(nums[i][j]);
            }
        }
        
        for(auto& it : sum) {
            result.insert(result.end(), it.begin(), it.end());
        }
        
        return result;
    }
};

File: Kotlin/1424_Diagonal_Traverse_II.kt
class Solution {
    fun findDiagonalOrder(nums: List<List<Int>>): IntArray {
        val result = mutableMapOf<Int, MutableList<Int>>()
        
        for(i in nums.indices) {
            for(j in nums[i].indices) {
               result.putIfAbsent(i + j, mutableListOf())
               result[i + j]?.add(0, nums[i][j])
           }
       }
       
       return result.map { it.value }.flatten().toIntArray()
    }
}